### DISCUSSION TITLE
Accidentally changed the owner of binary files from root to icub (username) in server laptop

### DISCUSSION THREAD

**Discussion by tasbolat1** [robotology/community#285] [Q&A] [answered]
Hello everyone,

During the work with iCub laptop (server), I  ran _sudo chown -R /_ command, which made icub (username) to be the owner of all possible binary files, including sudo. Hence, programs (which need root privileges) cannot be run anymore. Specifically, as a result
- cannot connect pc104 of icub
- can setup neither wire nor wireless connections
- cannot get intro RecoveryMode

When I run the _sudo_ command, I get the following error:
_/usr/bin/sudo must be owned by uid 0_

The iCub Laptop (server) came with iCub robot with pre-installed binaries set-up to work with robot.
After a long time search in the internet, I haven't found any solution for the problem. Could you help me with this issue?

Thanks in advance,
Tasbolat

**Comment by caicaixia** [comment]
:(

**Comment by lornat75** [comment]
can you try logging in as root? root should be able to re-assign correct ownership to files.

**Comment by tasbolat1** [comment]
@lornat75 Yes, I've tried, but no success.

To log in as a root from a terminal I need to have _sudo_ privileges. However, I cannot run _sudo_. I also cannot boot into recovery mode.

**Comment by mbrunettini** [comment]
Regarding the error with sudo command, it must be owned by root and have the suid bit setted
`chown root:root /usr/bin/sudo && chmod 4755 /usr/bin/sudo`
see [this link](https://askubuntu.com/questions/452860/usr-bin-sudo-must-be-owned-by-uid-0-and-have-the-setuid-bit-set)

**Comment by tasbolat1** [comment]
@mbrunettini 
I've checked the link, but problem is, again, I cannot log in as root. So, that solution doesn't work for this case.

**Comment by mbrunettini** [comment]
can you login as icub?

**Comment by tasbolat1** [comment]
yes

**Comment by mbrunettini** [comment]
did you tried to issue the above command as icub, then?

**Comment by tasbolat1** [comment]
Yes, I've tried. Have the following error:
_Permission denied_

**Comment by mbrunettini** [comment]
`ls -la /usr/bin/sudo` 
what returns?

**Comment by tasbolat1** [comment]
`-rwxr-xr-x 1 icub root 136808 lug 4 2017 /usr/bin/sudo`

**Comment by mbrunettini** [comment]
can you try the following?
` chmod 4755 /usr/bin/sudo`
`chown root:root /usr/bin/sudo`

**Comment by tasbolat1** [comment]
`chmod 4755 /usr/bin/sudo` can be executed.
However, `chown root:root /usr/bin/sudo` gives the following error:
`chown: changing ownership of /usr/bin/sudo: Operation not permitted`

**Comment by mbrunettini** [comment]
the most clean solution is to reinstall the machine: can you confirm this machine is the **server** laptop (i.e. the machine from where the pc104 mounts the robot code share)?
So I can point you to the right page on our wiki with all the instructions

**Comment by tasbolat1** [comment]
The laptop had arrived with iCub robot and set up to work with the iCub. Also username is `icub` and hostname is `icubsrv`. So, I confirm that the machine is a **server** laptop.

**Comment by mbrunettini** [comment, marked as answer]
Good, thanks.
Then, instructions are [here](https://wiki.icub.org/wiki/ICub_server_laptop_installation_instructions)
Please follow all the steps _carefully_

**Reply by smtbhd32** [reply]
not accesible

**Comment by drdanz** [comment]
You could also try logging in with a live distro (for example [SystemRescueCd](http://www.system-rescue-cd.org/)), mounting the hard drive and changing the ownership of the files from there...

**Comment by tasbolat1** [comment]
In the last stage, we have the following instruction:

> You have to use one of the Linux images we provide. This page contains some information about the Linux on the pc104, and important instructions for installing the startup scripts: The_Linux_on_the_pc104. 

There are two questions:
1. Do we need to re-install the Linux on the pc104?
2. If so, where can we get it?

**Comment by mbrunettini** [comment]
If it is already installed, of course not.

**Comment by tasbolat1** [comment]
I have a concern in the following part:

> git clone https://github.com/robotology/yarp.git

Should I do it as `root` or `icub`?

**Comment by mbrunettini** [comment]
icub
There is not the `sudo` command

**Comment by tasbolat1** [comment]
We followed the instruction given and could connect to the robot. We could also run `yarprobotinterface`.
However, when we run `yarpmotorgui`, we get the following error:
![screenshot from 2018-04-10 14-27-56](https://user-images.githubusercontent.com/4617666/38542333-cee30dd0-3ccb-11e8-82f0-70489ae22224.png)

**Comment by mbrunettini** [comment]
It is a different issue, I propose to close this one and to open a new one with a different topic.
Thanks

**Comment by tasbolat1** [comment]
Let me clear one thing:
We reinstall everything in the icub@icubsrv, but we didn't re-install or change anything inside the pc104.
So that means we downloaded and reinstalled all the codes in the server laptop (e.g. yarp, icub-main and so on) in /exports/code.  So I am wondering if the version of those code in pc104 is not match or link to this server laptop anymore.  Then it causes this problem?

My point is how to synchronize or map those source codes between the pc104 and server laptop?

**Comment by barbalberto** [comment]
Yes, you have to update YARP on the robot

**Comment by tasbolat1** [comment]
@barbalberto How shall we do? Do we need to get inside of the pc104 and git pull , then re-build it.
Or we can somehow to map from the server laptop?

**Comment by barbalberto** [comment]
You have to go inside the pc104
```
ssh on pc104
cd $YARP_DIR
git pull
cmake .. 
make -j2
```
then do the same on ICUB_DIR

**Comment by barbalberto** [comment]
And other folders you have in /usr/local/src/robot like icub-firmware-shared, in the order shown in the page http://wiki.icub.org/wiki/Compilation_on_the_pc104 from section compile yarp onward

**Comment by mbrunettini** [comment]
since the issue about the laptop is solved, i close this

**Comment by tasbolat1** [comment]
@barbalberto we updated yarp, icub-main and so other folders we mentioned. Now, we cannot run `yarprobotinterface`. My question is related to `ccmake` flags, which ones shall we make ON?
We followed the instructions you provided, however got the following errors:
![last](https://user-images.githubusercontent.com/4617666/38548000-d4d50f86-3cda-11e8-976d-5a1aad795e26.png)

**Comment by tasbolat1** [comment]
In

> Compile iCub,

 which flags should be ON? (We're iCubSingpapore01)
```
You need to enable (all versions of iCub):

 ENABLE_icubmod_dragonfly2
 ENABLE_icubmod_skinWrapper
 ENABLE_icubmod_xsensmtx

CAN robots:

 ENABLE_icubmod_sharedcan
 ENABLE_icubmod_canmotioncontrol
 ENABLE_icubmod_canBusAnalogSensor
 ENABLE_icubmod_canBusDoubleFTSensor
 ENABLE_icubmod_canBusSkin
 ENABLE_icubmod_canBusInertialMTB
 ENABLE_icubmod_canBusVirtualAnalogSensor
 ENABLE_icubmod_cfw2can
 ENABLE_icubmod_parametricCalibrator

ETH robots: enable also the following

 ENABLE_icubmod_embObjAnalogSensor
 ENABLE_icubmod_embObjMotionControl
 ENABLE_icubmod_embObjSkin
 ENABLE_icubmod_embObjVirtualAnalogSensor
 ENABLE_icubmod_parametricCalibratorEth

Cartesian controller:

 ENABLE_icubmod_cartesiancontrollerclient
 ENABLE_icubmod_cartesiancontrollerserver

Versions from iCub 1.1.1 enable also:

 ENABLE_icubmod_cfw2can
 ENABLE_icubmod_skinprototype

Previous versions (iCub 1.0 and 1.1.0):

 ENABLE_icubmod_pcan: iCub 1.0 and 1.1.0
 ENABLE_icubmod_ecan: iCub 1.1.0
```

**Comment by barbalberto** [comment]
Ok, you need also to update firmware on the boards.
Can you please tell us your YARP_ROBOT_NAME? Then we'll open a separated issue for that.

**Comment by tasbolat1** [comment]
iCubSingpapore01

**Comment by barbalberto** [comment]
I created issue here https://github.com/robotology/icub-support/issues/626
